piccolo/apps/schema/commands/generate.py

```diff
@@ -251,28 +267,134 @@ async def get_table_schema(
 
 
 async def get_foreign_key_reference(
-    table_class: t.Type[Table], constraint_name: str
-) -> t.Optional[str]:
+    table_class: t.Type[Table], constraint_name: str, constraint_schema: str
+) -> ConstraintTable:
     """
     Retrieve the name of the table that a foreign key is referencing.
     """
     response = await table_class.raw(
         (
-            "SELECT table_name "
+            "SELECT table_name, table_schema "
             "FROM information_schema.constraint_column_usage "
-            "WHERE constraint_name = {};"
+            "WHERE constraint_name = {} AND constraint_schema  = {};"
         ),
         constraint_name,
+        constraint_schema,
     )
     if len(response) > 0:
-        return response[0]["table_name"]
+        return ConstraintTable(
+            name=response[0]["table_name"], schema=response[0]["table_schema"]
+        )
     else:
-        return None
+        return ConstraintTable()
+
+
+def get_table_name(name: str, schema: str) -> str:
+    if schema == "public":
+        return name
+    return f"{schema}.{name}"
 
 
-async def get_output_schema(schema_name: str = "public") -> OutputSchema:
+async def create_table(
```

Comment:
I think the naming of this could be confusing - people might think it actually creates a table in the database.

Perhaps it can be called `create_table_class_from_db`?